src/identity/domain/Person.js

```diff
@@ -0,0 +1,314 @@
+ * @module domain
+ */
+
+/**
+ * Creates an domain layer person
+ * @class Person
+ * @memberof module:domain
+ * @param {Object} input The input object
+ * @param {string} input.id The uuid of the person
+ * @param {string} input.object The object type: person
+ * @param {string} input.userId The uuid of the user owner of the person
+ * @param {string} input.defaultAddressId The uuid of the default address
+ * @param {string} input.accountNumber The random account number
+ * @param {object} input.phoneNumber The phone number as defined in https://github.com/grantila/awesome-phonenumber via pn.toJSON()
+ * @param {module:domain.Address[]} input.addresses Array of addresses
+ * @param {module:domain.Setting} input.settings The settings for the person
+ * @param {string} input.avatar The avatar of the person (image)
+ * @param {number} input.pinCode [INT] The pin code to unlock the app
+ * @param {date} input.updatedAt The updated date
+ * @param {date} input.createdAt The created date
+ */
+const Person = attributes(
+  {
+    id: {
+      type: String,
+      required: false
+    },
+    object: {
+      type: String,
+      required: false,
+      default: 'person'
+    },
+    userId: {
+      type: String,
+      required: false
+    },
+    externalId: {
+      type: String,
+      required: false
+    },
+    profile: {
+      type: String,
+      required: false
+    },
+    status: {
+      type: String,
+      required: false
+    },
+    step: {
+      type: String,
+      required: false
+    },
+    identifier: {
+      type: String,
+      required: false
+    },
+    firstName: {
+      type: String,
+      required: false
+    },
+    lastName: {
+      type: String,
+      required: false
+    },
+    email: {
+      type: String,
+      required: false
+    },
+    phoneNumber: {
+      type: String,
+      required: false
+    },
+    addressA: {
+      type: String,
+      required: false
+    },
+    addressB: {
+      type: String,
+      required: false
+    },
+    city: {
+      type: String,
+      required: false
+    },
+    zip: {
+      type: String,
+      required: false
+    },
+    country: {
+      type: String,
+      required: false
+    },
+    dob: {
+      type: String,
+      required: false
+    },
+    taxId: {
+      type: String,
+      required: false
+    },
+    meta: {
+      type: Object,
+      required: false
+    },
+    images: {
+      type: Array,
+      itemType: 'Image',
+      required: false,
+      default: [],
+      empty: true,
+      nullable: true
+    },
+    updatedAt: {
+      type: Date,
+      required: false
+    },
+    createdAt: {
+      type: Date,
+      required: false
+    }
+  },
+  {
+    dynamics: {
+      Image: () => require('../image/Image')
+    }
+  }
+)(
+  class Person {
+    /**
+     * sets the userId field
+     * @memberof module:domain.Person
+     * @param {string} userId
+     * @returns {void}
+     */
+    addUserId (userId) {
```

Comment:
For this domain classes see the base domain https://github.com/charge-tech/chargejs/blob/master/src/domain.js#L13 
You can remove some functions from here that are common there, like `addUserId`